### ISSUE TITLE
add options to `require-passing-this`

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#123]
### Feature Description

The `require-passing-this` rule states that the `scope` property of the constructor must always be passed `this`, but in practice there are some cases where you ‘dare’ to pass something other than `this`. (Examples are shown below)

```typescript
const sample = new SampleConstruct(this, "Sample")
const bucket = new Bucket(sample, "Bucket")
```

Therefore, add the following options

```js
allowNonThisForNonScope: {
  type: "boolean",
  default: false
}
```

⬇️ example for eslint.config.js rule

```ts
export default [
  {
    // ... some configs
    rules: {
      "cdk/require-passing-this": ["error", { "allowNonThisForNonScope": true }]
    },
  },
];
```

### Invalid Code Examples

When `allowNonThisForNonScope` is true.
```typescript
import { Construct } from "constructs";
import { Bucket } from "aws-cdk-lib/aws-s3";

export class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    // ❌ 
    new Bucket(scope, "SampleBucket");
  }
}
```

### Valid Code Examples

When `allowNonThisForNonScope` is true.
```typescript
import { Construct } from "constructs";
import { Bucket } from "aws-cdk-lib/aws-s3";

export class MyConstruct extends Construct {
  constructor(scope: Construct, id: string) {
    super(scope, id);

    // ✅ 
    new Bucket(sample, "SampleBucket");
  }
}
```

### Rule Type

problem (possible errors)

### Implementation Details

- [ ] This rule is related to AWS CDK best practices
- [x] I'm willing to submit a PR for this feature
